units/stats_host_idle.go

```diff
@@ -67,6 +82,11 @@ func (j *collecHostIdleDataJob) Run() {
 		j.AddError(err)
 	}
 
+	if j.task == nil && j.TaskID != "" {
```

Comment:
I think line 60 means this can never be true.

Reply:
after creation, either: 
j.task is nil and j.TaskID is the empty 

-OR-

j.task is not nil and j.TaskID is not the empty string. 

-------------

the other case is that, if the jobs serialized into a remote job and round trips, j.task is nil (it wasn't originally) and j.TaskID has a value. In this case, we need to redo the query to get the task object, which is nil after runing through the bson processor.

Reply:
How does `j.TaskId` get set if the task is nil?

Reply:
because it isn't nil when we construct the job, but then we lose it when we serialize.